src/uwtools/drivers/driver.py

```diff
@@ -16,103 +11,70 @@
 
 class Driver(ABC):
     """
-    This base class provides the interface to methods used to read in a user-provided YAML
-    configuration file, convert it to the required config file, fix files, and namelist for a
-    forecast. Subsequent methods will be used to stage the input files and run the forecast.
-
-    Attributes
-    ----------
-    config_path : Path
-        The file path to the configuration file to be parsed.
-
-    Methods
-    -------
-    requirements()
-        Recursively parse config and platform files to determine and
-        fill in any dependencies.
-
-    resources()
-        Determine necessary task objects and fill in resource requirements of each.
-        Returns a Config object containing the HPC resources needed.
-
-    validate()
-        Validates the objects generated by the driver from the provided
-        config and platform files. If the --dry-run flag is provided, complete all
-        stages through validation and print results without running the forecast.
-
-
-    output()
-        Holds the knowledge for how to modify a list of output files and
-        stages them in the working directory.
-
-    job_card()
-        Turns the resources config object into a batch card for the
-        configured Task. Can be called from the command line on the front end
-        node to show the user what the job card would have looked like.
-
-    Notes
-    -------
-    Several functions such as create_model_config() and run() are relegated to
-    specific forecast drivers. This is because they are unique to the app being
-    run and will appropriately parse subsequent stages of the workflow.
+    An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: Optional[str] = None, log: Optional[Logger] = None):
+    def __init__(self, config_file: str, log: Optional[Logger] = None):
         """
-        Initialize the Forecast driver.
+        Initialize the driver.
         """
         self.log = log if log is not None else Logger()
-        if config_file is not None:
-            self.config_file = config_file
-            self._validate()
-        else:
-            self.log.info("No config file provided, available functions are limited.")
+        self._config_file = config_file
+        self._validate()
 
-    def _validate(self) -> bool:
+    # Public methods
+
+    @abstractmethod
+    def batch_script(self) -> None:
```

Comment:
Prefer "batch script" terminology to "job card". For example, compare the results returned from Google for query `"slurm" "batch script"` vs `"slurm" "job card"` (or replace `"slurm"` with `"pbs"`). While "job card" is not _totally_ unheard of, it seems clear to me that "batch script" is the much more standard terminology.

Reply:
Interestingly, both are used within SRW - it looks like "batch script" specifically refers to `rt.sh`,  which leads to somewhat mixed usage like `sbatch job_card`. Here job_card parses files like `fv3_slurm.IN_gaea` which is the sort of sbatch file we're trying to generate here. Looks like we're already at one of those conflicting terms things!

Reply:
So it doesn't get lost in the mix, let me @christinaholtNOAA on this one for an executive decision

Reply:
Let's choose "batch script" as the modern-era, single term we use to describe this thing.